src/Jackett.Common/Indexers/Corsarored.cs

```diff
@@ -1,256 +1,247 @@
-                queryCollection.RemoveAt(queryCollection.Count - 1); // remove last elem: page number 
-                queryCollection.Add("page", (++page).ToString());
-
-                var result = await SendAPIRequest(queryCollection);
-                try
-                {
-                    // this time is a jobject
-                    JObject json = (JObject)result;
-
-                    if (json["results"] == null)
-                        throw new Exception("Error invalid JSON response");
-
-                    // check number result
-                    if (((JArray)json["results"]).Count() == 0)
-                        break;
-
-                    foreach (var torrent in json["results"])
-                    {
-                        // add release
-                        releases.Add( makeRelease(torrent) );
-                    }
-                }
-                catch (Exception ex)
-                {
-                    OnParseError(result.ToString(), ex);
-                }
-
-            } while (page < MAX_SEARCH_PAGE_LIMIT);
-
-            return releases;
-        }
-
-        private ReleaseInfo makeRelease(JToken torrent)
-        {
-            var release = new ReleaseInfo();
-
-            release.Title = (String)torrent["title"];
-
-            //https://corsaro.red/details/E5BB62E2E58C654F4450325046723A3F013CD7A4
-            release.Comments = new Uri(SiteLink + "details/" + (string)torrent["hash"]);
-            release.Guid = release.Comments;
-            //release.Link = release.Comments;
-
-            release.PublishDate = DateTime.Now;
-            if (torrent["last_updated"] != null)
-                release.PublishDate = DateTime.Parse((string)torrent["last_updated"]);
-
-            // TODO: don't know how to map this cats..
-            int cat = (int)torrent["category"];
-            release.Category = MapTrackerCatToNewznab(cat.ToString());
-
-            if (torrent["size"] != null)
-                release.Size = (long)torrent["size"];
-
-            release.Grabs = (long)torrent["completed"];
-
-            release.Description = (string)torrent["category"] + " " + (string)torrent["description"];
-
-            /*
-            RageID = copyFrom.RageID;
-            Imdb = copyFrom.Imdb;
-            TMDb = copyFrom.TMDb;
-            */
-
-            release.Seeders = (int)torrent["seeders"];
-            release.Peers = release.Seeders + (int)torrent["leechers"];
-
-            release.InfoHash = (string)torrent["hash"];
-            release.MagnetUri = new Uri((string)torrent["magnet"]);
-
-            /*
-            MinimumRatio = copyFrom.MinimumRatio;
-            MinimumSeedTime = copyFrom.MinimumSeedTime;
-            */
-
-            release.DownloadVolumeFactor = 0;
-            release.UploadVolumeFactor = 1;
-
-            return release;
-        }
-    }
-}
+	public class Corsarored : BaseWebIndexer
+	{
+		private const int MaxSearchPageLimit = 8; // 1page 25 items, 200
```

Comment:
Calling 8 times for each search seems too much for me. Maybe 4 instead?

Reply:
Yeah, I just left it where the last guy had it. Wasn't sure.